subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -4501,26 +4501,20 @@ static inline int length_req_rsp_recv(struct ll_conn *conn, memq_link_t *link,
 	if (/* Local idle, and Peer request then complete the Peer procedure
 	     * with response.
 	     */
-	    ((conn->llcp_length.req == conn->llcp_length.ack) &&
-	     (pdu_rx->llctrl.opcode == PDU_DATA_LLCTRL_TYPE_LENGTH_REQ)) ||
+	    ((conn->llcp_length.req == conn->llcp_length.ack) && tx) ||
```

Comment:
The criteria is that opcode equals PDU_DATA_LLCTRL_TYPE_LENGTH_REQ, not that tx!=NULL
Even though tx!=NULL when the opcode is correct, this is not immediately obvious, you need to scroll back in the code.
IMHO the original was easier to follow. 
Are there significant reductions in code-size and/or execution time?

Reply:
Compare and branch on zero take less instructions and CPU cycles, in compare to load two values compare it and branch set of instructions.

Before simplification:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      251093 B         1 MB     23.95%
            SRAM:       33652 B       256 KB     12.84%
        IDT_LIST:         136 B         2 KB      6.64%
```
After simplification:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      251029 B         1 MB     23.94%
            SRAM:       33652 B       256 KB     12.84%
        IDT_LIST:         136 B         2 KB      6.64%
```